main/src/init/file_init.hpp

```diff
@@ -135,9 +162,9 @@ cstone::Box<typename HydroData::RealType> restoreHydroData(const std::string& h5
     errors |= fileutils::readH5PartField(h5_file, "z", d.z.data());
     errors |= fileutils::readH5PartField(h5_file, "h", d.h.data());
     errors |= fileutils::readH5PartField(h5_file, "m", d.m.data());
-    errors |= fileutils::readH5PartField(h5_file, "temp", d.temp.data());
+    errors |= initUTemp(h5_file, rank,d);
```

Comment:
It would be better to convert any left-over checkpoint HDF5 files from `u` to `temp` with a python script and simply require `temp` as input. The version here is not that useful as it won't be able to adapt to different equations of state. For a more general EOS with u != cv * temp, you will have both `temp` and `u`. Here you'll try to load both and throw an error one level above in the test-case specific initializer if the file didn't contain the required information.